src/arrayop.d

```diff
@@ -177,8 +177,7 @@ extern (C++) Expression arrayOp(BinExp e, Scope* sc)
     /* Append deco of array element type
      */
     buf.writestring(e.type.toBasetype().nextOf().toBasetype().mutableOf().deco);
-    char* name = buf.peekString();
-    Identifier ident = Identifier.idPool(name);
+    auto ident = Identifier.idPool(buf.getSlice());
```

Comment:
You should at least call it `peekSlice` so it's clear that the buffer still owns and frees the underlying string.
Extracting the string and transfering it's ownership to idPool would save a pointless copy, maybe add a flag to idPool whether it needs to copy the string.